libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -12,4 +13,20 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       _applicationId: applicationId,
     });
   }
+
+  async create(data: Partial<IntegrationEntity>): Promise<IntegrationEntity> {
+    const existingIntegration = await this.findOne({
+      _applicationId: data._applicationId,
+      providerId: data.providerId,
+      channel: data.channel,
+    });
+    if (!existingIntegration) {
+      return await super.create(data);
+    }
+
+    throw new HttpException(
+      HttpStatusCode.UNPROCESSABLE_ENTITY,
+      'Duplicate key - One application may not have two providers of the same channel type'
+    );
```

Comment:
I've just thought about something haven't caught before. Should we introduce the concept of HTTP in the dal layer? I mean that it's a very specific client implementation to accessing data that can be accessed with graphql, HTTP, cli, queue system, etc...

I think that the "dal" should be client agnostic and introducing the HTTP concepts and error codes kind of breaks this rule. What do you think about that? @ainouzgali @djabarovgeorge

Reply:
And to bring it closer to the point, should we just throw a regular `Error` object here? Or create a new wrapper "DalException" or something ?

Reply:
I'm am glad you noticed I do believe that the DAL should not be familiar with HTTP exceptions, due to the reason we use MongoDB being a  DBMS on the data storage layer. We do not validate duplication keys on other layers however I do want to give the other layers some more information. Do you think we could just transfer the error message between layers and send ApiException?
Regarding your question, I think if we won't send any additional data regular `Error` should be more the enough.

Reply:
On other hand maybe adding something wrapper that will help us differentiate between our errors and unknown errors will be a good idea.

Reply:
Let's create a DalException error object extending the default error. Wdyt?

Reply:
Sure I'm in.